tcms/static/js/tcms_actions.js

```diff
@@ -1392,20 +1391,25 @@ function exportCase(url, form, table) {
 
 var printableCases = exportCase;
 
-function validateIssueID(bugSystemId, bugId) {
-  if (bugSystemId == 1) {
-    var result = /^\d{1,7}$/.test(bugId);
-    if (!result) {
-      window.alert(default_messages.alert.invalid_bug_id);
-    }
-    return result;
-  } else if (bugSystemId == 2) {
-    var result = /^[A-Z0-9]+-\d+$/.test(bugId);
-    if (!result) {
-      window.alert(default_messages.alert.invalid_jira_id);
-    }
-    return result;
+function validateIssueID(bugRegExp, bugId) {
+  // if bugRegExp is empty string then all input is valid!
+  if (!bugRegExp) {
+    return true;
+  }
+
+  try {
+    var bug_re = new RegExp(bugRegExp);
+  // catch syntax errors in regular expressions
+  } catch(err) {
+    window.alert(err.name + ': ' + err.message);
```

Comment:
Hmmm, in this case we have a problem with the regexp compiling. Isn't JavScript going to provide the most descriptive error message, as to why the regex won't compile?  Sure 'Your bug validation regexp is not correct' looks better but it doesn't help the user figure out what's wrong. Regular expressions are notoriously hard to debug. Can you tell me which one is better:
`SyntaxError: Invalid regular expression: /$?/: Nothing to repeat` or a generic message?